### PR TITLE
[WIP] timesheet activity report

### PR DESCRIPTION
**Pull request by xie8899**

### CODE REVIEW

**Comment by elicoidal** [comment]
@xie8899 It seems something is wrong in this PR: there are 2 modules and I think wc connector should be in OCA (please check with @noahzaozao)
Anyway, please only push one module at the time.

**Comment by xie8899** [comment]
done

**Comment by elicoidal** [comment]
The version inthe repo seems newer than your current one. Can you recheck ?

**Comment by noahzaozao** [comment]
@elicoidal I'm confirmed we're using the superman version to override the odoo-addons , so I saw some fields are different , may be the superman one is current we're using.

**Comment by elicoidal** [comment]
Possible. One thing we can do ir probably copy the README files from this branch into the Superman's addon.

File: timesheet_activity_report/__init__.py [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
Comment (noahzaozao) [inline timesheet_activity_report/__init__.py]: header

File: timesheet_activity_report/README.rst [hunk 2]
```diff
@@ -4,48 +4,4 @@
 
 =========================
 Timesheet Activity Report
-=========================
```
Comment (noahzaozao) [inline timesheet_activity_report/README.rst:7]: use the odoo-addons version of readme.rst

**Comment by noahzaozao** [comment]
LGTM
@elicoidal @seb-elico please review this PR , thanks.

**Comment by elicoidal** [comment]
LGTM
name should be business_requirement_activity_report

**Comment by seb-elico** [comment]
@elicoidal I don't agree to change the name. This is the TS report (a.k.a. "TS completion report"), which goes up to the BR if it exists. But it also includes TS from issues and TS input directly in the TS activities. The BR is just an info attached to the TS. It's not the same report as the Project completion report (which we could rename BR completion report indeed) that gives the state of a BR/project, even if there's currently no TS attached (and gathers the TS spent on each task).
TL;DR: DO NOT RENAME IT

**Comment by seb-elico** [comment]
In the future, we might cut this module into several modules, depending on the dependencies (`hr_timesheet`, `project_timesheet`, `project_issue_sheet`, `business_requirement`)

**Comment by elicoidal** [comment]
I have a problem with this module as it depends on the BR category.
Unfortunately It is difficult to cut/inherit properly a SQL statement.
@seb-elico

**Comment by xie8899** [comment]
@elicoidal @seb-elico @noahzaozao Renamed.

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+class TimesheetReport(osv.Model):
+    """ Timesheet Activities Report """
+    _name = "timesheet.activity.report"
+    _auto = False
+    _description = "Timesheet Activities Report"
+    _rec_name = 'activity_name'
+
+    _columns = {
+        'id': fields.integer(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please fit this on 1 line?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+            help="Type is used to separate Tasks, Issues and\
+                Timesheets input directly"),
+        'description': fields.char(
+            'Description', readonly=True),
+        'hours': fields.float(
+            'Time spent',
+            digits=(16, 2),
+            readonly=True,
+            help="Time spent on timesheet"),
+        'user_id': fields.many2one(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please fit this on 1 line?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            'res.users', 'User', readonly=True),
+        'product_id': fields.many2one(
+            'product.product', 'Product', readonly=True),
+        'date': fields.date(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please fit this on 1 line?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            help="Activity Stage"),
+        'account_id': fields.many2one(
+            'account.analytic.account', 'Analytic account', readonly=True),
+        'activity_id': fields.char(
+            'Activity id', readonly=True, help="Task id or Issue id"),
+        'activity_name': fields.char(
+            'Activity name', readonly=True, help="Task name or Issue name"),
+        'task_categ_id': fields.many2one(
+            'task.category', 'Task cat.', readonly=True, help="Task category"),
+        'br_id': fields.many2one(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please regroup to fit on 3 lines instead of 5?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+            'business.requirement',
+            'Bus. requ.',
+            readonly=True,
+            help="Business requirement"),
+        'partner_id': fields.many2one(
+            'res.partner', 'Customer', readonly=True),
+        'project_categ_id': fields.many2one(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please regroup to fit on 3 lines instead of 5?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        'activity_type': fields.selection(
+            [
+                ('task', 'Task'),
+                ('issue', 'Issue'),
+                ('timesheet', 'Timesheet'),
+            ],
+            'Type',
+            readonly=True,
+            help="Type is used to separate Tasks, Issues and\
+                Timesheets input directly"),
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Are you sure this is a string?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    _columns = {
+        'id': fields.integer(
+            'Id', readonly=True),
+        'activity_type': fields.selection(
+            [
+                ('task', 'Task'),
+                ('issue', 'Issue'),
+                ('timesheet', 'Timesheet'),
+            ],
+            'Type',
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please fit these 2 parameters on 1 line?

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+            readonly=True,
+            help="Type is used to separate Tasks, Issues and\
+                Timesheets input directly"),
+        'description': fields.char(
+            'Description', readonly=True),
+        'hours': fields.float(
```
Comment (seb-elico) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: Can you please regroup to fit on 2 lines instead of 5?

**Comment by xie8899** [comment]
@elicoidal @seb-elico @noahzaozao fixed all comments

File: business_requirement_activity_report/report/timesheet_activity_report.py (hunk 6 above)
Comment (xie8899) [reply business_requirement_activity_report/report/timesheet_activity_report.py]: done

File: business_requirement_activity_report/README.rst [hunk 11]
```diff
@@ -24,17 +24,15 @@ Bug Tracker
 Bugs are tracked on `GitHub Issues <https://github.com/Elico-Corp/odoo/issues>`_.
 In case of trouble, please check there if your issue has already been reported.
 If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
-`here <https://github.com/Elico-Corp/odoo/issues/new?body=module:%20timesheet_activity_report%0Aversion:%20{8.0}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+`here <https://github.com/Elico-Corp/odoo/issues/new?body=module:%20business_requirement_activity_report%0Aversion:%20{8.0}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline business_requirement_activity_report/README.rst]: remove this lline

**Comment by elicoidal** [comment]
LGTM

File: business_requirement_activity_report/report/timesheet_activity_report.py [hunk 12]
```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+#   2016 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp.osv import fields, osv
+from openerp import tools
+
+
+class TimesheetReport(osv.Model):
```
Comment (martintorresvm) [inline business_requirement_activity_report/report/timesheet_activity_report.py]: @xie8899 is totally necessary to use old API or could be done on new API so the possible porting it will be easier?

Comment (seb-elico) [reply business_requirement_activity_report/report/timesheet_activity_report.py]: @victormartinelicocorp I developed it in old API in the first place because I copied/pasted the code from another old API module and I didn't know how to port it to the new API. I think it should be quite easy to port, but this is not the priority.
@elicoidal do you confirm we can keep the old API?

File: business_requirement_activity_report/README.rst (hunk 11 above)
Comment (seb-elico) [reply business_requirement_activity_report/README.rst]: @xie8899 reminder

File: business_requirement_activity_report/report/timesheet_activity_report.py (hunk 12 above)
Comment (elicoidal) [reply business_requirement_activity_report/report/timesheet_activity_report.py]: Agree

**Comment by noahzaozao** [comment]
@seb-elico Could you review this PR , thanks

**Comment by seb-elico** [comment]
LGMT
@noahzaozao @xie8899 there are conflicts, please solve so that we can merge

**Comment by noahzaozao** [comment]
@seb-elico I've solved conflict , but don't merge it , since the module name has been changed.

**Comment by seb-elico** [comment]
LGTM
@noahzaozao clear, let's not merge it until we uninstall it. Then we merge it and re-install it!
@elicoidal @victormartinelicocorp @xuan139 FYI

**Comment by elicoidal** [comment]
@seb-elico @noahzaozao What is the status here? Close or merge?

**Comment by seb-elico** [comment]
@elicoidal definitely we need to merge it eventually

@noahzaozao @victormartinelicocorp since the BR modules have been improved a lot, please make sure this PR is based on the latest version. My previous LGTM is maybe obsolete!

@noahzaozao we use this module in Elico8 right? If we resolve the conflicts and merge, what's the impact right now? We first uninstall the module, merge the PR, pull the code and re-install the module with the new name, right?
